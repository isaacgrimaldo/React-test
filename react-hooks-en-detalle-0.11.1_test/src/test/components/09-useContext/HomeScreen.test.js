import React from 'react';
import '@testing-library/jest-dom';
import {mount} from 'enzyme';

import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas del archivo HomeScreen.js ', () => {
      
    const user = {
        name: 'Isaac',
        email: 'Isaac@gmail.com',
    };

    const wrapper = mount(  //Como queremos renderizar tambien el componente hijo se usa el mount y no shallow 
           <UserContext.Provider
              value = {{ 
                    user  
              }}
           >
              <HomeScreen />
           </UserContext.Provider>
      );

      test('Debe de renderizar correctamente el componente', () => {
           expect(wrapper).toMatchSnapshot();
      });

});