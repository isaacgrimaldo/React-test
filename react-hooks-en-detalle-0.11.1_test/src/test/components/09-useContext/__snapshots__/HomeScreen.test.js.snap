// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del archivo HomeScreen.js  Debe de renderizar correctamente el componente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Isaac",
   "email": "Isaac@gmail.com"
}
  </pre>
</div>
`;
